File: app/src/main/java/com/dhc/mvp/net/RoomSampleActivity.java
package com.dhc.mvp.net;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.dhc.library.base.BaseApplication;
import com.dhc.library.base.XDaggerActivity;
import com.dhc.library.utils.AppContext;
import com.dhc.mvp.R;
import com.dhc.mvp.dao.AppDatabase;
import com.dhc.mvp.di.DiHelper;
import com.dhc.mvp.modle.bean.GankItemBean;
import com.dhc.mvp.presenter.NetTestPresenter;
import com.dhc.mvp.presenter.RoomTestPresenter;
import com.dhc.mvp.presenter.RxCacheTestPresenter;
import com.dhc.mvp.presenter.contract.INetTestContract;

import org.w3c.dom.Text;

import java.util.List;

import io.reactivex.functions.Consumer;

/**
 * @creator：denghc(desoce)
 * @updateTime：2018/8/23 下午2:55
 * @description：使用 Room示例
 */
public class RoomSampleActivity extends XDaggerActivity<RoomTestPresenter> implements INetTestContract
        .IView<List<GankItemBean>> {

    private TextView title;
    private TextView content;

    @Override
    public int getLayoutId() {
        return R.layout.activity_rxcache_sample;
    }

    @Override
    public void initEventAndData(Bundle savedInstanceState) {
        title = $(R.id.tv_title);
        title.setText("Url : http://gank.io/api/random/data/福利/1");
        content = $(R.id.tv_content);
        mPresenter.getRandomGirl();//调用方法请求接口
    }

    @Override
    public void initInject(Bundle savedInstanceState) {
        DiHelper.getActivityComponent(getActivityModule()).inject(this);
    }

    @Override
    public void success(List<GankItemBean> data) {
        content.setText(getString(R.string.data_f,data.get(0).toString()));
        mPresenter.loadAllByIds(new String[]{data.get(0).get_id()}).subscribe(new
                                                                          Consumer<List<GankItemBean>>() {
            @Override
            public void accept(List<GankItemBean> gankItemBeans) throws Exception {
                ((TextView )$(R.id.tv_database)).setText("数据库数据: "+gankItemBeans.get(0).toString());
            }
        });
    }

    @Override
    public void failure(String code, String msg) {

    }
}

File: app/src/main/java/com/dhc/mvp/net/RoomSampleActivity.kt
package com.dhc.mvp.net

import android.os.Bundle
import androidx.activity.viewModels
import androidx.lifecycle.Observer

import com.dhc.library.base.activity.BaseVmDbActivity
import com.dhc.library.base.viewmodel.parseState
import com.dhc.mvp.R
import com.dhc.mvp.databinding.ActivityRxcacheSampleBinding
import com.dhc.mvp.state.HomeViewModel
import com.dhc.mvp.viewmodel.RequestDbViewModel
import kotlinx.android.synthetic.main.activity_rxcache_sample.*
import kotlinx.android.synthetic.main.activity_rxcache_sample.tv_content
import kotlinx.android.synthetic.main.activity_rxcache_sample.tv_title

/**
 * @creator：denghc(desoce)
 * @updateTime：2018/8/23 下午2:55
 * @description：使用 Room示例
 */
class RoomSampleActivity : BaseVmDbActivity<HomeViewModel, ActivityRxcacheSampleBinding>() {

    private val viewModel: RequestDbViewModel by viewModels()


    override fun createObserver() {
        viewModel.run {
            randomGirlData.observe(this@RoomSampleActivity, Observer {
                parseState(it, onSuccess = { data ->
                    tv_content!!.text = getString(R.string.data_f, data!![0].toString())
                    viewModel.loadAllByIds(arrayOf(data[0]._id))
                }, onError = { })
            })
            dbRandomGirlData.observe(this@RoomSampleActivity, Observer {
                parseState(it, { data -> tv_database!!.text = "数据库数据: " + data[0].toString() })
            })
        }
    }

    override fun initView(savedInstanceState: Bundle?) {
        tv_title!!.text = "Url : http://gank.io/api/random/data/福利/1"
        viewModel!!.getRandomGirl()//调用方法请求接口
    }
}
